selectrum.el

```diff
@@ -643,9 +643,42 @@ behavior."
         ;; Skip updates.
         (setq-local selectrum--skip-updates-p t)))))
 
+(defun selectrum--minibuffer-matchstring-bounds ()
+  "Return bounds for current matchstring.
+The current matchstring may be surrounded by prefix and suffix."
+  (let* ((input (minibuffer-contents))
+         (pt (- (point) (minibuffer-prompt-end)))
+         (bounds (completion-boundaries
+                  (substring input 0 pt)
+                  minibuffer-completion-table
+                  minibuffer-completion-predicate
+                  (if (and minibuffer-completing-file-name
+                           (looking-back "/" (1- (point)))
+                           (looking-at "/"))
+                      ""
+                    (substring input pt))))
+         (start (+ (minibuffer-prompt-end)
+                   (car bounds)))
+         (end (+ (minibuffer-prompt-end)
+                 (+ pt (cdr bounds)))))
+    (cons start end)))
+
 (defun selectrum--get-full (candidate)
-  "Get full form of CANDIDATE by inspecting text properties."
+  "Get full form of CANDIDATE."
   (or (get-text-property 0 'selectrum-candidate-full candidate)
```

Comment:
Would it be possible to kick out the selectrum-candidate-full property then or is there are deeper reason? Compatibility, caching?

Reply:
It's part of the public API, internally there is currently no more use for it.

Reply:
Cool! This means the get-full function is now the equivalent to the corresponding function in embark/marginalia? If that's the case, this is great since this ensures compatibility. Is it possible to deprecate the property somehow? I would be all for reducing the API surface of selectrum if possible!

Reply:
It should be similar with some internal handling for selectrum. I will wait with breaking changes until @raxod502 has time to look into selectrum again but I think it could be removed evenetually.

Reply:
Cool! selectrum-candidate-full does not appear to be widely used https://github.com/search?q=selectrum-candidate-full&type=code

Reply:
Sorry for the delay. In a perfect world I think we'd deprecate the property first, and later remove it. But, given that you can literally search and see how many (few) people are using it, I think it would be fine to just remove it immediately, which would make the next release a major version number bump.

Reply:
Although, if we get rid of this text property, what is the replacement? Does one need to use a custom completion table function? That seems a bit user-unfriendly. It might be nice to support both ways, the simple but limited way as well as the more advanced but API-compliant way.

Reply:
It can be a useful feature but I think when using it with `completing-read` a command written with this would probably break when used without Selectrum as the property is able to change the return value.

Reply:
Yes, that seems right to me. I suppose instead we should have only properties that change the display of candidates, and not ones that change the return values.

Reply:
That sounds good to me, too. In that case we can remove `selectrum-candidate-full` we don't use it internally any more :)

Reply:
SGTM!